File: ui/base-tool/src/main/java/org/cru/godtools/base/tool/activity/BaseArticleActivity.java
package org.cru.godtools.base.tool.activity;

import org.cru.godtools.xml.model.Manifest;

import androidx.annotation.NonNull;

public abstract class BaseArticleActivity extends BaseSingleToolActivity {
    public BaseArticleActivity(final boolean immersive) {
        super(immersive);
    }

    public BaseArticleActivity(final boolean immersive, final boolean requireTool) {
        super(immersive, requireTool);
    }

    @Override
    protected boolean isSupportedType(@NonNull final Manifest.Type type) {
        return type == Manifest.Type.ARTICLE;
    }
}

File: ui/base-tool/src/main/kotlin/org/cru/godtools/base/tool/activity/BaseArticleActivity.kt
package org.cru.godtools.base.tool.activity

import androidx.annotation.LayoutRes
import androidx.annotation.StringRes
import androidx.databinding.ViewDataBinding
import org.cru.godtools.shared.tool.parser.model.Manifest
import org.cru.godtools.tool.R

abstract class BaseArticleActivity<B : ViewDataBinding> protected constructor(
    @LayoutRes contentLayoutId: Int,
    requireTool: Boolean
) : BaseSingleToolActivity<B>(contentLayoutId, requireTool, Manifest.Type.ARTICLE) {
    protected constructor(@LayoutRes contentLayoutId: Int) : this(contentLayoutId, true)

    @get:StringRes
    override val shareLinkMessageWithQrCodeRes get() = R.string.share_tool_message_article
//    override val shareLinkMessageRes get() = R.string.share_tool_message_article
}
